lib/_http_client.js

```diff
@@ -639,7 +639,9 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
   }
 
   this.once('socket', function(sock) {
-    sock.setTimeout(msecs, emitTimeout);
+    sock.once('connect', function() {
+      this.setTimeout(msecs, emitTimeout);
```

Comment:
in the off chance that someone updates this later to an arrow function, the use of `this` internally may be slightly problematic. Perhaps make it more explicit using `sock.setTimeout(msec, emitTimeout);` instead?